dmqtt/signals.py

```diff
@@ -8,16 +8,32 @@
 
 logger = logging.getLogger(__name__)
 
-connect = django.dispatch.Signal(providing_args=["userdata", "flags", "rc"])
-message = django.dispatch.Signal(providing_args=["userdata", "msg"])
-
+try:
+    connect = django.dispatch.Signal(providing_args=["userdata", "flags", "rc"])
+    message = django.dispatch.Signal(providing_args=["userdata", "msg"])
+except TypeError:
+    # for Django < 3.0
+    connect = django.dispatch.Signal()
+    message = django.dispatch.Signal()
+
+def convert_wildcards(mqtt_topic):
+    """Convert MQTT wildcards to regex pattern.
+    '#' becomes '.*' (multi-level)
+    '+' becomes '[^/]+' (single-level)
+    """
+    pattern = re.escape(mqtt_topic)
+    pattern = pattern.replace('\\#', '.*')
+    pattern = pattern.replace('\\+', '[^/]+')
+    return '^' + pattern + '$'
```

Comment:
```suggestion
def convert_wildcards(mqtt_topic) -> re.Pattern:
    """Convert MQTT wildcards to regex pattern.
    '#' becomes '.*' (multi-level)
    '+' becomes '[^/]+' (single-level)
    """
    pattern = re.escape(mqtt_topic)
    pattern = pattern.replace("\\#", ".*")
    pattern = pattern.replace("\\+", "[^/]+")
    return re.compile(f"^{pattern}$")
```

We can do the compiling of the pattern here once, and return it as a small optimization.